src/utils/arguments.ts

```diff
@@ -351,6 +415,25 @@ export class Args {
     return this;
   }
 
+  /**
+   * Adds an array.
+   *
+   * @remarks
+   * If the type of the values of the array is not native type, this will serialize the pointers, which is certainly not
+   * what you want. You can only serialize properly array of native types or array of `Serializable` object.
+   *
+   * @see {@link addSerializableObjectArray}
+   *
+   * @param arg - the argument to add
+   * @returns the modified Arg instance
+   */
+  addNativeTypeArray<T extends TypedArrayUnit>(arg: any[], type: T): Args {
```

Comment:
Imho the function signature shoud be:
`addNativeTypeArray<T>(arg: T[]): Args {`
Then you should be able to handle different types like this 
```
  switch (idof<T>()) {
    case idof<i8[]>(): // ArrayI8;
    case idof<u8[]>(): // ArrayU8;
    case idof<i16[]>(): // ArrayI16;
    case idof<u16[]>(): // .ArrayU16;
    ```
etc...
 It have the benefit to infer the intern type of the array instead of explicitely setting it everytime. and thus make the code cleaner.
 Not tested of course.. so its only a improvement suggestions!